Admin/Action.php

```diff
@@ -108,58 +131,87 @@ public function getPermissions()
     }
 
     /**
-     * @param string[] $permissions
+     * @return Action[]
      */
-    public function setPermissions($permissions)
+    public function getActions()
     {
-        $this->permissions = $permissions;
+        return $this->actions;
     }
 
     /**
-     * @return mixed
+     * @param array $actions
      */
-    public function getRoute()
+    public function setActions($actions)
     {
-        return $this->route;
+        $this->actions = $actions;
     }
 
     /**
-     * @param mixed $route
+     * @param Action $action
      */
-    public function setRoute($route)
+    public function addAction(Action $action)
     {
-        $this->route = $route;
+        $this->actions[$action->getName()] = $action;
     }
 
     /**
-     * @return mixed
+     * @return array
      */
-    public function getParameters()
+    public function getFilters()
     {
-        return $this->parameters;
+        return $this->filters;
     }
 
     /**
-     * @param mixed $parameters
+     * @param array $filters
      */
-    public function setParameters($parameters)
+    public function setFilters($filters)
+    {
+        $this->filters = $filters;
+    }
+
+    public function addFilter(Filter $filter)
     {
-        $this->parameters = $parameters;
+        $this->filters[] = $filter;
     }
 
     /**
```

Comment:
Not it is not deprecated but it is not used yet neither. Waiting for FilterBundle from @VincentChalnot